src/forms/signUp.jsx

```diff
@@ -9,24 +9,32 @@ import Typography from "@mui/material/Typography";
 import Container from "@mui/material/Container";
 import { ThemeProvider } from "@mui/material/styles";
 import theme from "../theme";
+import { useDispatch } from "react-redux";
 import Btn from "../components/button";
+import { signUp } from "../services/tableDataServices";
+import { sign_up } from "../redux/actions/action";
 
 export default function SignUp() {
   const [userName, setUserName] = useState("");
   const [password, setPassword] = useState("");
   const [email, setEmail] = useState("");
-  const handleSubmit = (event) => {
+  const dispatch = useDispatch();
+
+  const handleSubmit = async (event) => {
     event.preventDefault();
-    // const data = new FormData(event.currentTarget);
-    // console.log({
-    //   email: data.get("email"),
-    //   password: data.get("password")
-    // });
+    const newUser = {
+      userName: userName,
+      email: email,
+      password: password
+    };
+    const DataApi = await signUp(newUser);
+    dispatch(sign_up(DataApi));
```

Comment:
Data must be dispatched if API response is true. Means there should be a condition checking if API returned correct response.